mkosi/__init__.py

```diff
@@ -1446,7 +1449,12 @@ def mount_image(
     with complete_step("Mounting image"):
 
         if root_dev is not None:
-            mount_loop(args, root_dev, root, root_read_only)
+            if args.usr_only:
+                # In UsrOnly mode let's have a bind mount at the top so that umount --recursive works nicely later
+                mount_bind(root)
+                mount_loop(args, root_dev, os.path.join(root, "usr"), root_read_only)
+            else:
+                mount_loop(args, root_dev, root, root_read_only)
```

Comment:
Not sure I follow here, for usr_only, is_generated_root() returns True so mount_image() will always be called with root_dev set to None right which means we get the bnid mount from the else case of the parent if statement.

Reply:
There are two calls of mount_image(). The second one is where this matters: where we already built almost evertyhing and now need to mount it in order to be able to call dracut to generate the right kernel for it. In that scond call root_dev will be non-None, and that's where this matters

Reply:
(the second one is the one with the lambda stuff)